FBSimulatorControl/Interactions/FBSimulatorInteraction+Setup.m

```diff
@@ -94,6 +94,34 @@ - (instancetype)authorizeLocationSettingForApplication:(FBSimulatorApplication *
   return [self authorizeLocationSettings:@[application.bundleID]];
 }
 
+- (instancetype)overrideWatchDogTimerForApplication:(NSString *)bundleID withTimeout:(NSTimeInterval)timeout
+{
+  NSParameterAssert(bundleID);
+  NSParameterAssert(timeout);
+
+  return [self interactWithShutdownSimulator:^ BOOL (NSError **error, FBSimulator *simulator) {
+    NSString *simulatorRoot = simulator.device.dataPath;
+
+    NSString *preferencesDirectory = [simulatorRoot stringByAppendingPathComponent:@"Library/Preferences"];
+    NSError *innerError = nil;
+    if (![NSFileManager.defaultManager createDirectoryAtPath:preferencesDirectory withIntermediateDirectories:YES attributes:nil error:&innerError]) {
+      return [FBSimulatorError failBoolWithError:innerError description:@"Failed to create preferences dir" errorOut:error];
+    }
+
+    NSString *springboardSettingsPath = [preferencesDirectory stringByAppendingPathComponent:@"com.apple.springboard.plist"];
+    NSMutableDictionary *springboardSettings = [NSMutableDictionary dictionaryWithContentsOfFile:springboardSettingsPath] ?: [NSMutableDictionary dictionary];
+    springboardSettings[@"FBLaunchWatchdogExceptions"] =
+    @{
+      bundleID: @(timeout),
```

Comment:
I think we should make this an array of bundle ids so that you can set the watchdog timer for multiple apps at once, without having to re-write the file every time.